fabtests/ubertest/uber.c

```diff
@@ -652,82 +431,109 @@ static int ft_fw_server(void)
 
 		results[ft_fw_result_index(ret)]++;
 
-	} while (!ret || ret == FI_EIO || ret == FI_ENODATA);
+	} while (!ret || ret == FI_EIO || ret == FI_ENODATA || ret == FT_SKIP);
 
 	return ret;
 }
-
-static int ft_client_child(void)
+static int ft_client_setup(struct fi_info *hints, struct fi_info *info)
 {
-	struct fi_info *hints, *info;
-	int ret, result, server_ready = 0;
+	int ret = 0;
+	ret = ft_recv_test_info();
```

Comment:
Is it possible for the server to return ENODATA and then send the test_info to the client who does not return an error and then begins to run the test? What happens in that case?

Reply:
I'm not sure where an ENODATA would occur, but if there is an error sending the sockets data, I don't think it would reach the test to start it.